crates/stages/src/id.rs

```diff
@@ -98,17 +98,20 @@ impl StageId {
     }
 
     /// Get the last committed progress of this stage.
-    pub fn get_progress<'db>(&self, tx: &impl DbTx<'db>) -> Result<Option<BlockNumber>, DbError> {
+    pub fn get_checkpoint<'db>(
+        &self,
+        tx: &impl DbTx<'db>,
+    ) -> Result<Option<StageCheckpoint>, DbError> {
         tx.get::<SyncStage>(self.0.to_string())
     }
 
     /// Save the progress of this stage.
-    pub fn save_progress<'db>(
+    pub fn save_checkpoint<'db>(
```

Comment:
love this change, now I no longer have to wrap my head around what progress means :)